### PR TITLE
Ft gif comment

### PR DESCRIPTION
**Pull request by Ebunwadi**

### CODE REVIEW

File: database/db.sql [hunk 1]
```diff
@@ -19,9 +19,21 @@ CREATE TABLE gifs (
...
     public_id VARCHAR(255) NOT NULL,
     created_on timestamp with time zone NOT NULL,
+    is_flagged BOOLEAN DEFAULT false,
     user_id serial NOT NULL,
     FOREIGN KEY (user_id)
     REFERENCES users (id)
 );
 
+CREATE TABLE gif_comment (
+    comment_id SERIAL PRIMARY KEY,
```
Comment (courage173) [inline database/db.sql]: use id as the pry key

File: test/gif.test.js [hunk 2]
```diff
@@ -11,7 +12,9 @@ chai.should();
 
 chai.use(chaiHttp);
 
-const token = jwt.sign({ userId: 101, isAdmin: true, email: 'my@gmail.com' }, process.env.JWT_SECRET);
+const token = jwt.sign({ userid: 90, isAdmin: true, email: 'hea@gmail.com' }, process.env.JWT_SECRET);
+const tokenn = jwt.sign({ userid: 100, isAdmin: false, email: 'headies@gmail.com' }, process.env.JWT_SECRET);
```
Comment (courage173) [inline test/gif.test.js:16]: why do we have double n

File: controllers/gif.comment.controller.js [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+      data: {
+        message: 'comment successfully created',
+        createdOn,
+        gifTitle: gif.rows[0].title,
+        comment,
+        userId,
+      },
+    });
+  } catch (error) {
+    console.log(error);
```
Comment (courage173) [inline controllers/gif.comment.controller.js]: remove console

File: controllers/gif.comment.controller.js [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+  const { gifId } = req.params;
+  const gifComment = await pool.query('SELECT * FROM gif_comment WHERE gif_id = $1', [gifId]);
+  if (gifComment.rows.length === 0) {
+    return res.status(404).json({
+      status: 'error',
+      error: 'there is no comment on this gif',
+    });
+  }
+
+  if (gifComment.rows[0].is_flagged === true) {
```
Comment (courage173) [inline controllers/gif.comment.controller.js]: you don't need to add this condition. they should be able to delete any comment

**Review by courage173 — CHANGES_REQUESTED** [review]

File: database/db.sql (hunk 1 above)
Comment (Ebunwadi) [reply database/db.sql]: i should change comment_id to id?

File: test/gif.test.js (hunk 2 above)
Comment (Ebunwadi) [reply test/gif.test.js:16]: to generate another token to test if a user is not an admin

File: controllers/gif.comment.controller.js (hunk 3 above)
Comment (Ebunwadi) [reply controllers/gif.comment.controller.js]: okay

File: controllers/gif.comment.controller.js (hunk 4 above)
Comment (Ebunwadi) [reply controllers/gif.comment.controller.js]: okay

**Review by courage173 — APPROVED** [review]

**Pull request merged**